go/tasks/plugins/k8s/kfoperators/common/common_operator.go

```diff
@@ -11,15 +11,33 @@ import (
 	flyteerr "github.com/flyteorg/flyteplugins/go/tasks/errors"
 	"github.com/flyteorg/flyteplugins/go/tasks/logs"
 	pluginsCore "github.com/flyteorg/flyteplugins/go/tasks/pluginmachinery/core"
+	mpiOp "github.com/kubeflow/common/pkg/apis/common/v1"
 	commonOp "github.com/kubeflow/tf-operator/pkg/apis/common/v1"
 	v1 "k8s.io/api/core/v1"
 )
 
 const (
 	TensorflowTaskType = "tensorflow"
+	MPITaskType        = "mpi"
 	PytorchTaskType    = "pytorch"
 )
 
+func ExtractMPICurrentCondition(jobConditions []mpiOp.JobCondition) (mpiOp.JobCondition, error) {
```

Comment:
gah this is annoying you have to duplicate all these tasks for mpi job conditions